### PR TITLE
Chore/rails 51

### PR DESCRIPTION
**Pull request by tsubik**
First step to upgrade backend to newer Rails version.

### CODE REVIEW

File: app/controllers/v1/users_controller.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ def update
     end
 
     def current
-      user = User.find(context[:current_user])
+      user = User.find(context[:current_user].id)
```
Comment (santostiago) [inline app/controllers/v1/users_controller.rb:27]: mmm... this worked before? :O

File: app/controllers/v1/users_controller.rb [hunk 2]
```diff
@@ -62,7 +62,7 @@ def update_params
           :locale,
         )
       # better in this way to keep actioncontroller::parameters object, this rails version has some bugs
-      p.deep_transform_keys!(&:underscore)
+      p.transform_keys!(&:underscore)
```
Comment (santostiago) [inline app/controllers/v1/users_controller.rb:65]: why the change?

File: app/models/observation.rb [hunk 3]
```diff
@@ -83,13 +83,13 @@ class Observation < ApplicationRecord
   attr_accessor :user_type
 
   belongs_to :country,        inverse_of: :observations
```
Comment (santostiago) [inline app/models/observation.rb:85]: shouldn't we also add the `optional: true` to the country? I'm not sure if it's mandatory in the observations tool.

File: app/models/operator_document_history.rb [hunk 4]
```diff
@@ -31,11 +31,11 @@
...
 
-  belongs_to :operator, optional: false
-  belongs_to :required_operator_document, -> { with_archived }, required: true
+  belongs_to :operator
+  belongs_to :required_operator_document, -> { with_archived }
   belongs_to :fmu , -> { with_deleted },  optional: true
   belongs_to :user, optional: true
-  belongs_to :document_file, optional: :true
+  belongs_to :document_file, optional: true
   belongs_to :operator_document, -> { with_deleted }
```
Comment (santostiago) [inline app/models/operator_document_history.rb:39]: mmm... I have a feeling this should be optional. 
Isn't there a case in which the operator document can be removed, but the history remains? (maybe not...)

File: app/models/photo.rb [hunk 5]
```diff
@@ -18,7 +18,7 @@
 class Photo < ApplicationRecord
   mount_base64_uploader :attachment, PhotoUploader
 
-  belongs_to :user, inverse_of: :photos
+  belongs_to :user, inverse_of: :photos, optional: true # TODO: I think this is model is not used could be removed
```
Comment (santostiago) [inline app/models/photo.rb:21]: Me too. Did you check the server to see if there are any files there?

File: spec/models/operator_document_spec.rb [hunk 6]
```diff
@@ -188,14 +189,15 @@
       end
     end
 
-    describe '#insure_unity' do
+    describe '#ensure_unity' do
```
Comment (santostiago) [inline spec/models/operator_document_spec.rb:192]: I made this mistake billions of times....

File: spec/factories/operator_document_histories.rb [hunk 7]
```diff
@@ -28,13 +28,14 @@
 #
 
 FactoryBot.define do
-  factory :operator_document_history do
+  factory :operator_document_history, class: OperatorDocumentCountryHistory, aliases: [:operator_document_country_history] do
```
Comment (santostiago) [inline spec/factories/operator_document_histories.rb:31]: I just didn't understand this part (and the similar ones below)

**Review by santostiago — APPROVED** [review]

File: app/controllers/v1/users_controller.rb (hunk 1 above)
Comment (tsubik) [reply app/controllers/v1/users_controller.rb:27]: Yes, in 5.0 it worked, although I've never seen that method used in that way, first time here. In 5.1 that's ArgumentError telling to use id.

File: app/controllers/v1/users_controller.rb (hunk 2 above)
Comment (tsubik) [reply app/controllers/v1/users_controller.rb:65]: ActiveController::Parameters has no longer all Hash object methods, so there is no `deep_transform_keys` let me dig into history why key transform was added here. Comment is interesting saying there was some bug in the framework, maybe it's gone in 5.1 version, so I could just remove that method.

File: app/models/observation.rb (hunk 3 above)
Comment (tsubik) [reply app/models/observation.rb:85]: It's ok, no nulls in the database, so I guess it's required or there are no cases to be optional.

File: app/models/operator_document_history.rb (hunk 4 above)
Comment (tsubik) [reply app/models/operator_document_history.rb:39]: As document is soft-deleted then all those cases are bad data cases as it does not make sense to operator_document to be optional. We don't have any foreign keys in db, so one of the next PRs will add them and we have to deal with bad data.

File: app/models/photo.rb (hunk 5 above)
Comment (tsubik) [reply app/models/photo.rb:21]: Nothing in the database and nothing on the disk. I will remove that model.

File: spec/factories/operator_document_histories.rb (hunk 7 above)
Comment (tsubik) [reply spec/factories/operator_document_histories.rb:31]: Well, if possible better to stay away from STI as it has many shortcomings. Here there was some problem with factories if I have not set class I think. By default, I made `operator_document_history` factory to create country document. Maybe there should be no `operator_document_history` factory at all, only country and fmu ones, but I did not want to change all the specs.

File: app/controllers/v1/users_controller.rb (hunk 2 above)
Comment (tsubik) [reply app/controllers/v1/users_controller.rb:65]: ok, I need that `transform_keys` here, but it does not have to be deep, so all good.

File: app/models/photo.rb (hunk 5 above)
Comment (tsubik) [reply app/models/photo.rb:21]: but in the next PR

Comment (tsubik) [reply app/models/photo.rb:21]: to keep them more tidy.
